common/client.go

```diff
@@ -336,5 +355,9 @@ func (rc *RestClient) GetServiceConfig(rootServiceUrl string, svc Service) (*Ser
 	if err != nil {
 		return nil, err
 	}
+	// Save the credential from the client in the resulting service config --
```

Comment:
Ok misleading comment here. The flow is like this:

To initialize a service we need its config from root. To do so, we use a default RestClient with credentials supplied on command-line. Since we are just initializing, we know we first should auth, and we'll get a token which the client will save. We would like to pass that token into this newly initialized service so that it can use it when it creates a RestClient of its own.